### ISSUE TITLE
CMake install not compatible with TriBITS install

### ISSUE THREAD

**Issue by sslattery** [kokkos/kokkos-kernels#869]
Per

https://github.com/kokkos/kokkos-kernels/blob/f64b1c57f52989600fe75c1dad2fdcd0d4323528/src/CMakeLists.txt#L417

the CMake install will ultimately append the Kokkos namespace to the target such that `Kokkos::kokkoskernels` may be linked by a downstream user. If instead someone wanted to use a kokkos-kernels build from Trilinos the result is a target named `kokkoskernels` without the namespace and thus the target is not found resulting in config errors such as:
```
CMake Error at unit_test/CMakeLists.txt:128 (add_executable):
  Target "ParticleInterpolation_test_OPENMP" links to target
  "Kokkos::kokkoskernels" but the target was not found.  Perhaps a
  find_package() call is missing for an IMPORTED target, or an ALIAS target
  is missing?
```
in codes that are expecting the namespace. The target names should be consistent between different installation options to allow for both a CMake or Trilinos install of kokkos-kernels to be used with a given application.

For reference, the Kokkos install in Trilinos does get installed with the namespace (e.g. `Kokkos::kokkoscore`) so this should be doable.

**Comment by sslattery** [comment]
@lucbv @srajama1 any ideas on this one?

**Comment by srajama1** [comment]
Yes, this is doable. Let me check what the CMake experts recommend.

**Comment by lucbv** [comment]
@sslattery I can take a look at this later this week. Out of curiosity, in what context are you using this?
Actually if that is used by open code, on top of satisfying my curiosity this would help me reproduce the problem.
Otherwise a small reproducer would be appreciated.

**Comment by sslattery** [comment]
Hi @lucbv - We have a package downstream of Cabana that uses KokkosKernels. Some applications that use this downstream package have a Trilinos dependency while others do not and therefore we want to support both GitHub-based and Trilinos-based builds of Kokkos/KokkosKernels. 

The Kokkos install creates the same target structure regardless of whether the install is from the main git repo or through a Trilinos install but KokkosKernels is different.

The code is open and we are going to be populating a repo on GitHub soon.

Basically though I want to be able to do this in my top level `CMakeLists.txt` - 
```
find_package(KokkosKernels REQUIRED)
```
 and then do this when I build my library - 
```
target_link_libraries( ${lib_name} Kokkos::kokkos Kokkos::kokkoskernels )
```
That works fine through GitHub CMake but I need to strip the `Kokkos` namespace from the KokkosKernels target if I use Trilinos

**Comment by lucbv** [comment]
I have generated a reproducer, I have dug a little into our CMake logic but still trying to find the best way to fix this
